src/main/webapp/app/exercises/shared/structured-grading-criterion/grading-instructions-details/grading-instructions-details.component.ts

```diff
@@ -497,4 +504,78 @@ export class GradingInstructionsDetailsComponent implements OnInit, AfterContent
             this.exercise.gradingInstructions = text;
         }
     }
+
+    /**
+     * Switches edit mode
+     */
+    switchMode() {
+        this.showEditMode = !this.showEditMode;
+    }
+
+    /**
+     * Updates given grading instruction in exercise
+     *
+     * @param gradingInstruction needs to be updated
+     * @param criterion includes instruction needs to be update
+     */
+    updateGradingInstruction(instruction: GradingInstruction, criterion: GradingCriterion) {
+        const criterionIndex = this.exercise.gradingCriteria!.indexOf(criterion);
+        const instructionIndex = this.exercise.gradingCriteria![criterionIndex].structuredGradingInstructions.indexOf(instruction);
+        this.exercise.gradingCriteria![criterionIndex].structuredGradingInstructions![instructionIndex] = instruction;
+    }
+
+    /**
+     * Updates credits of the GradingInstruction.
+     *
+     * @param gradingInstruction needs to be updated
+     * @param criterion includes instruction needs to be update
+     */
+    setCredits($event: any, instruction: GradingInstruction, criterion: GradingCriterion) {
+        instruction.credits = $event.target.value;
+        this.updateGradingInstruction(instruction, criterion);
+    }
```

Comment:
for all those methods:
- `setCredits`, `setGradingScale`, `setGradingInstructionDescription`, `setFeedback`, `setUsageCount`
can't you combine those?
E.g. by a method:
```
/**
 * Updates changed properties of the GradingInstruction.
 *
 * @param gradingInstruction that needs to be updated
 * @param criterion that includes the instruction needs to be updated
 * @param property that is updated
 */
updateGradingInstructionProperty($event: any, instruction: GradingInstruction, criterion: GradingCriterion, property: ENUMFORPROPERTY): {
    (switch ENUMFORPROPERTY) {
        case ENUMFORPROPERTY.GradingScale: {
            instruction.gradingScale = $event.target.value;
        }
        case ENUMFORPROPERTY.Credits: {
            instruction.credits = $event.target.value;
        }
        //... one case for each field that can be Updated
    }
    this.updateGradingInstruction(instruction, criterion);
}
````
You would have to create an enum, I just used `ENUMFORPROPERTY` as placeholder.
In the html template you should then call:
`(change)="updateGradingInstructionProperty($event, instruction, criterion, ENUMFORPROPERTY.X)"`
or maybe you can distinguish the cases due to information that is included in the event you are passing into the method, I am not sure about this.

If you prefer staying with these separate methods, it's also fine, it's just a suggestion.